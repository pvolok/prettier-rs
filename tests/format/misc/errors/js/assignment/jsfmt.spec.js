run_spec(
  {
    importMeta: import.meta,
    snippets: ["({}) = x;"],
  },
  ["babel", "babel-ts", "acorn", "espree", "meriyah"],
);

run_spec(
  {
    importMeta: import.meta,
    snippets: ["class Foo extends ((let)[0] = 1) {}"],
  },
  ["acorn", "espree", "meriyah"],
);

run_spec(
  {
    importMeta: import.meta,
    snippets: ["export default ((let)[0] = 1);"],
  },
  ["espree", "meriyah"],
);
