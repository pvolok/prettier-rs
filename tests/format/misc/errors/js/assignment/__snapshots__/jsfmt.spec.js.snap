// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snippet: #0 [acorn] format 1`] = `
"Assigning to rvalue (1:1)
> 1 | ({}) = x;
    | ^"
`;

exports[`snippet: #0 [acorn] format 2`] = `
"The keyword 'let' is reserved (1:21)
> 1 | class Foo extends ((let)[0] = 1) {}
    |                     ^"
`;

exports[`snippet: #0 [babel] format 1`] = `
"Invalid parenthesized assignment pattern. (1:1)
> 1 | ({}) = x;
    | ^"
`;

exports[`snippet: #0 [babel-ts] format 1`] = `
"Invalid parenthesized assignment pattern. (1:1)
> 1 | ({}) = x;
    | ^"
`;

exports[`snippet: #0 [espree] format 1`] = `
"Assigning to rvalue (1:1)
> 1 | ({}) = x;
    | ^"
`;

exports[`snippet: #0 [espree] format 2`] = `
"The keyword 'let' is reserved (1:21)
> 1 | class Foo extends ((let)[0] = 1) {}
    |                     ^"
`;

exports[`snippet: #0 [espree] format 3`] = `
"The keyword 'let' is reserved (1:18)
> 1 | export default ((let)[0] = 1);
    |                  ^"
`;

exports[`snippet: #0 [meriyah] format 1`] = `
"Invalid left-hand side in assignment (1:6)
> 1 | ({}) = x;
    |      ^"
`;

exports[`snippet: #0 [meriyah] format 2`] = `
"The identifier 'let' must not be in expression position in strict mode (1:24)
> 1 | class Foo extends ((let)[0] = 1) {}
    |                        ^"
`;

exports[`snippet: #0 [meriyah] format 3`] = `
"The identifier 'let' must not be in expression position in strict mode (1:21)
> 1 | export default ((let)[0] = 1);
    |                     ^"
`;
